src/instruction.js

```diff
@@ -20,13 +21,19 @@ class Instruction {
         if (this.#plane.isLanded() === true) {
             throw this.#landError;
         }
+        if (this.#airport.checkWeather() === 'stormy') {
+            throw this.#weatherError;
+        }
         this.#airport.landPlane(this.#plane);
     }
 
     takeOffPlane() {
         if (this.#plane.isLanded() === false) {
             throw this.#takeOffError;
         }
+        if (this.#airport.checkWeather() === 'stormy') {
+            throw this.#weatherError;
+        }
```

Comment:
Even Better If:

- I could have used a method to handle the logic of checking for errors, instead of having them within the methods that should only be doing one thing e.g. landing or take-off a plane. This would make my code look cleaner, and it would be easier to maintain.